src/test/java/fi/aalto/cs/intellij/activities/VersionCheckerActivityTest.java

```diff
@@ -0,0 +1,42 @@
+package fi.aalto.cs.intellij.activities;
+
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.startup.StartupActivity;
+import fi.aalto.cs.intellij.common.BuildInfo;
+import fi.aalto.cs.intellij.notifications.APlusNotifications;
+import org.hamcrest.CoreMatchers;
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+public class VersionCheckerActivityTest {
+
+  @Test
+  public void testRunActivityWithPreReleaseVersion() {
+    int[] numberOfCalls = { 0 };
+
+    StartupActivity activity = new VersionCheckerActivity(
+        new BuildInfo.Version(0, 22, 315),
+        (notification, project) -> {
+          numberOfCalls[0]++;
+          Assert.assertNull(project);
+          Assert.assertThat(notification,
+              CoreMatchers.instanceOf(APlusNotifications.BetaVersionWarning.class));
+          Assert.assertEquals("0.22.315",
+              ((APlusNotifications.BetaVersionWarning) notification).getVersion().toString());
+        });
+
+    activity.runActivity(Mockito.mock(Project.class));
+
+    Assert.assertEquals(1, numberOfCalls[0]);
```

Comment:
I understand why and wo functional interfaces a kind of `Mockito.verify()` could be tried, but I can't figure out if it's pluggable here

Reply:
I'd guess we could also use Mockito's dependency injection to mock, say, IntelliJ's `Notifications.Bus.notify(Nofification)`. Personally, I like abstract interfaces more because it makes it explicit which entries of the framework each class use (forcing coder to actually think: *should* this class use this feature of the framework) and also provides greater independence of implementation details of the framework.

With functional interfaces (which I haven't use that much before this project), this approach even looks clean in the code.